### PR TITLE
feat: add scroll for merchant, org, and profile dropdowns

### PR DESCRIPTION
**Pull request by kanikabansal-juspay**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

This PR fixes an issue where the merchant, org, and profile dropdowns were missing scroll functionality. The scroll has been added to improve usability when the dropdown content exceeds the visible area.

Changes
- Added scroll functionality to the merchant, org, and profile dropdowns.
- Ensured that overflow is handled correctly .
- Added borders to avoid merging of profile dropdown with the background.

**Before**

Screenshot 2024-10-03 at 11 57 01 AM

Screenshot 2024-10-03 at 12 02 45 PM

**After** 

Screenshot 2024-10-03 at 4 48 35 PM

Screenshot 2024-10-03 at 4 52 12 PM

**After Updates :**

Screenshot 2024-10-04 at 12 00 13 PM

## Motivation and Context

To improve the user experience when interacting with the dropdown menus

## How did you test it?

Working as described in description by adding multiple profiles and merchants.

## Where to test it?

- [x] INTEG
- [x] SANDBOX
- [ ] PROD

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Comment by Riddhiagrawal001** [comment]
- Can you increase the width of the profile dropdown as in figma 
- Please verify that other dropdown components are not affected due to this as you have changed in SelectBox component

**Review by sagarnaikjuspay — DISMISSED** [review]

File: src/components/SelectBox.res [hunk 1]
```diff
@@ -1169,8 +1170,26 @@ module RenderListItemInBaseRadio = {
...
+
+        .sidebar-scrollbar::-webkit-scrollbar-track {
+          display: none;
+        }
+      }
+    `
     <>
-      <div className=""> {dropdownList} </div>
+      <div className={`${customScrollStyle}  sidebar-scrollbar`} >
+       <style> {React.string(sidebarScrollbarCss)} </style> {dropdownList} </div>
```
Comment (gitanjli525) [inline src/components/SelectBox.res:1192]: Can you add a condition to show scrollbar (the <style> tag)  only if `customScrollStyle` is not empty string?
Same for sidebar-scrollbar class, add that class only if `customScrollStyle` is present

File: src/screens/OMPSwitch/ProfileSwitch.res [hunk 2]
```diff
@@ -134,9 +134,10 @@ let make = () => {
     }
   }
 
-  let customPadding = "px-1 py-1"
+  let customPadding = ""
```
Comment (gitanjli525) [inline src/screens/OMPSwitch/ProfileSwitch.res:137]: can you remove this variable as it is not used anymore

File: src/screens/OMPSwitch/OMPSwitchHelper.res [hunk 3]
```diff
@@ -24,7 +24,7 @@ module ListBaseComp = {
 
 module AddNewMerchantProfileButton = {
   @react.component
-  let make = (~user, ~setShowModal, ~customPadding="", ~customStyle="", ~customHRTagStyle="") => {
+  let make = (~user, ~setShowModal, ~customPadding="", ~customStyle="", ~customHRTagStyle="",~addProfileBtnStyle="") => {
```
Comment (gitanjli525) [inline src/screens/OMPSwitch/OMPSwitchHelper.res:27]: can you rename this variable to more generic name as this component is being used for merchant also ?

File: src/screens/OMPSwitch/MerchantSwitch.res [hunk 4]
```diff
@@ -162,7 +162,7 @@ let make = () => {
   let customHRTagStyle = "border-t border-blue-830"
   let customPadding = "py-1 w-full"
   let customStyle = "w-56 text-gray-200 bg-blue-840 dark:bg-black hover:bg-popover-background-hover hover:text-gray-100 !w-full"
-
+  let customScrollStyle = "max-h-72 overflow-scroll scrollbar-thin px-1 pt-1 "
```
Comment (gitanjli525) [inline src/screens/OMPSwitch/MerchantSwitch.res:165]: `scrollbar-thin` class is not needed as you have already mentioned it in `sidebarScrollbarCss` in SelectBox

File: src/screens/OMPSwitch/ProfileSwitch.res (hunk 2 above)
Comment (kanikabansal-juspay) [reply src/screens/OMPSwitch/ProfileSwitch.res]: Yes removed the unused variable.

File: src/screens/OMPSwitch/OMPSwitchHelper.res (hunk 3 above)
Comment (kanikabansal-juspay) [reply src/screens/OMPSwitch/OMPSwitchHelper.res]: Updated the variable name.

File: src/screens/OMPSwitch/MerchantSwitch.res (hunk 4 above)
Comment (kanikabansal-juspay) [reply src/screens/OMPSwitch/MerchantSwitch.res]: Removed the redundant scrollbar-thin class.

**Review by gitanjli525 — DISMISSED** [review]

**Review by Riddhiagrawal001 — APPROVED** [review]

File: src/components/SelectBox.res [hunk 5]
```diff
@@ -1169,8 +1170,35 @@ module RenderListItemInBaseRadio = {
...
+
+        .sidebar-scrollbar::-webkit-scrollbar-track {
+          display: none;
+        }
+      }
+    `
+    let (className, styleElement) = switch customScrollStyle {
+    | None => ("", React.null)
+    | Some(style) => (
+        style ++ " sidebar-scrollbar",
```
Comment (JeevaRamu0104) [inline src/components/SelectBox.res:1193]:
```suggestion
        `${style}  sidebar-scrollbar`,
```

**Review by PritishBudhiraja — APPROVED** [review]
Looks good :')

**Review by JeevaRamu0104 — APPROVED** [review]

**Pull request merged**